server/src/main/java/org/elasticsearch/search/fetch/ShardFetchRequest.java

```diff
@@ -42,10 +41,10 @@
 
     private ScoreDoc lastEmittedDoc;
 
-    public ShardFetchRequest(ShardSearchContextId contextId, IntArrayList list, ScoreDoc lastEmittedDoc) {
+    public ShardFetchRequest(ShardSearchContextId contextId, ArrayList<Integer> docIds, ScoreDoc lastEmittedDoc) {
         this.contextId = contextId;
-        this.docIds = list.buffer;
-        this.size = list.size();
+        this.docIds = docIds.stream().mapToInt(i -> i).toArray();
+        this.size = docIds.size();
```

Comment:
This size is no longer needed as a member, since now the array is sized correctly. Before the buffer from the hppc type could have been larger.